### PR TITLE
User profile page

### PR DESCRIPTION
**Pull request by nmking22**
- [x] done

User Story 19, User Profile Show Page

As a registered user
When I visit my profile page
Then I see all of my profile data on the page except my password
And I see a link to edit my profile data

- [x] done

User Story 20, User Can Edit their Profile Data

As a registered user
When I visit my profile page
I see a link to edit my profile data
When I click on the link to edit my profile data
I see a form like the registration page
The form is prepopulated with all my current information except my password
When I change any or all of that information
And I submit the form
Then I am returned to my profile page
And I see a flash message telling me that my data is updated
And I see my updated information

- [x] done

User Story 21, User Can Edit their Password

As a registered user
When I visit my profile page
I see a link to edit my password
When I click on the link to edit my password
I see a form with fields for a new password, and a new password confirmation
When I fill in the same password in both fields
And I submit the form
Then I am returned to my profile page
And I see a flash message telling me that my password is updated

- [x] done

User Story 22, User Editing Profile Data must have unique Email address

As a registered user
When I attempt to edit my profile data
If I try to change my email address to one that belongs to another user
When I submit the form
Then I am returned to the profile edit page
And I see a flash message telling me that email address is already in use

- [ ]

New Issue: User attribute update works in testing, but is not updating on localhost/database

### CODE REVIEW

File: app/controllers/password_controller.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    current_user.update(password_params)
+    redirect_to '/profile', notice: 'Your password has been updated.'
+  end
+
+  private
+
+  def password_params
+    params.permit(:password, :password_confirmation)
+  end
+end
```
Comment (BJSherman80) [inline app/controllers/password_controller.rb:15]: good use of private params it makes the code much more simple to read

File: spec/features/users/show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    click_on 'Edit Profile'
+
+    expect(current_path).to eq('/profile/edit')
+
+    fill_in :email, with: @existing_user.email
+    click_button 'Update Profile'
+
+    expect(current_path).to eq('/profile/edit')
+    expect(page).to have_content('That email address is already in use.')
+
```
Comment (BJSherman80) [inline spec/features/users/show_spec.rb:95]: Great job testing sad path

**Review by BJSherman80 — APPROVED** [review]

**Review by the-color-bliu — APPROVED** [review]

**Comment by the-color-bliu** [comment]
Everything looks good, we still have the one issue to resolve where the db isn't updating but otherwise merge is approved.

File: app/controllers/password_controller.rb (hunk 1 above)
Comment (helloeduardo) [reply app/controllers/password_controller.rb:15]: Nice solution! Creative use of creating a new controller for those paths

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -1,9 +1,12 @@
...
+    <%= "Name: #{current_user.name}" %><br>
+    <%= "Address: #{current_user.address}" %><br>
+    <%= "City: #{current_user.city}" %><br>
+    <%= "State: #{current_user.state}" %><br>
+    <%= "Zip Code: #{current_user.zip}" %><br>
+    <%= "Email: #{current_user.email}" %><br>
 </ul></h3>
+
+<%= link_to "Edit Profile", "/profile/edit" %>
+<%= link_to "Edit Password", "/password/edit" %>
```
Comment (helloeduardo) [inline app/views/users/show.html.erb:12]: Like the two separate links to two different paths

**Review by helloeduardo — APPROVED** [review]
Looks good, I'll try to see if I can replicate the issue you were having with updating password on my local

**Pull request merged**